packages/patternfly-4/react-core/src/components/ClipboardCopy/ClipboardCopy.tsx

```diff
@@ -19,34 +20,89 @@ const clipboardCopyFunc = (event, text) => {
   clipboard.removeChild(el);
 };
 
-export const ClipboardCopyVariant = {
-  inline: 'inline',
-  expansion: 'expansion'
-};
+export enum ClipboardCopyVariant {
+  inline = 'inline',
+  expansion = 'expansion'
+}
+
+export interface ClipboardCopyState {
+  text: string | number; 
+  expanded: boolean;
+  copied: boolean;
+}
+
+export interface ClipboardCopyProps extends Omit<React.HTMLProps<HTMLDivElement>, 'onChange'> {
+  /** Additional classes added to the clipboard copy container. */
+  className?: string;
+  /** Tooltip message to display when hover the copy button */
+  hoverTip?: string;
+  /** Tooltip message to display when clicking the copy button */
+  clickTip?: string;
+  /** Custom flag to show that the input requires an associated id or aria-label. */
+  textAriaLabel?: string;
+  /** Custom flag to show that the toggle button requires an associated id or aria-label. */
+  toggleAriaLabel?: string;
+  /** Flag to show if the input is read only. */
+  isReadOnly?: boolean;
+  /** Adds Clipboard Copy variant styles. */
+  variant?: typeof ClipboardCopyVariant | 'inline' | 'expansion'; 
+  /** Copy button popover position. */
+  position?: OneOf<typeof PopoverPosition, keyof typeof PopoverPosition>;
+  /** Maximum width of the tooltip (default 150px). */
+  maxWidth?: string;
+  /** Delay in ms before the tooltip disappears. */
+  exitDelay?: number;
+  /** Delay in ms before the tooltip appears. */
+  entryDelay?: number;
+  /** Delay in ms before the tooltip message switch to hover tip. */
+  switchDelay?: number;
+  /** A function that is triggered on clicking the copy button. */
+  onCopy?: (event: React.ClipboardEvent<HTMLDivElement>, text?: React.ReactNode) => void;
+  /** A function that is triggered on changing the text. */
+  onChange?: (text?: string | number) => void;
+  /** The text which is copied. */
+  children?: React.ReactNode;
+}
 
-class ClipboardCopy extends React.Component {
-  constructor(props) {
+export class ClipboardCopy extends React.Component<ClipboardCopyProps, ClipboardCopyState> {
+  timer = null as NodeJS.Timer;
+  constructor(props: ClipboardCopyProps) {
     super(props);
-    this.timer = null;
     this.state = {
-      text: this.props.children,
+      text: this.props.children as string | number,
       expanded: false,
       copied: false
     };
   }
 
-  expandContent = () => {
+  static defaultProps = {
+    hoverTip: 'Copy to clipboard',
+    clickTip: 'Successfully copied to clipboard!',
+    isReadOnly: false,
+    variant: ClipboardCopyVariant.inline,
```

Comment:
Didn't we have to use the actual value her rather than the enum?  I thought there was some reason for that @redallen.

Reply:
The actual value makes future refactoring to remove the enum easier. That's all.